File: lib/domain/field/dtos/field_detail_input.rb
# frozen_string_literal: true

module Domain
  module Field
    module Dtos
      class FieldDetailInput
        attr_reader :field_id, :farm_id

        # @param field_id [Integer, String]
        # @param farm_id [Integer, String, nil] route scope for HTML redirect on failure
        def initialize(field_id:, farm_id: nil)
          @field_id = field_id.to_i
          @farm_id = farm_id.nil? ? nil : farm_id.to_i
        end
      end
    end
  end
end

File: crates/agrr-domain/src/field/dtos/field_detail_input.rs
/// Ruby: `Domain::Field::Dtos::FieldDetailInput`
#[derive(Debug, Clone, PartialEq)]
pub struct FieldDetailInput {
    pub field_id: i64,
    pub farm_id: Option<i64>,
}

impl FieldDetailInput {
    pub fn new(field_id: i64) -> Self {
        Self {
            field_id,
            farm_id: None,
        }
    }

    pub fn with_farm_id(field_id: i64, farm_id: i64) -> Self {
        Self {
            field_id,
            farm_id: Some(farm_id),
        }
    }
}
